docs/reference/advanced_configuration.rst

```diff
@@ -9,66 +9,66 @@ Full configuration options:
 
 .. code-block:: yaml
 
-    fos_user:
-        db_driver:        orm # can be orm or mongodb (support is also available within FOSUser for couchdb, propel but none is given for SonataUserBundle)
-        firewall_name:    main
-        user_class:       Application\Sonata\UserBundle\Entity\User
-
-        group:
-            group_class:  Application\Sonata\UserBundle\Entity\Group
-
-        profile:
-            # Authentication Form
-            form:
-                type:               fos_user_profile
-                name:               fos_user_profile_form
-                validation_groups:  [Authentication] # Please note : this is not the default value
-
     sonata_user:
         security_acl: false
-        manager_type: orm      # can be orm or mongodb
+        firewall_name: 'admin'
+        manager_type: 'orm' # can be orm or mongodb
 
         table:
-            user_group: "my_custom_user_group_association_table_name"
+            user_group: 'my_custom_user_group_association_table_name'
 
         impersonating:
-            route:                page_slug
-            parameters:           { path: / }
+            route: page_slug
+            parameters: { path: / }
 
-        class:                  # Entity Classes
-            user:               Application\Sonata\UserBundle\Entity\User
-            group:              Application\Sonata\UserBundle\Entity\Group
+        class: # Entity Classes
+            user: 'App\Entity\SonataUserUser'
+            group: 'App\Entity\SonataUserGroup'
```

Comment:
Previously the entity was in the UserBundle, now it have to be created by the user ?
Cant/Shouldnt we provide a User class ?

Reply:
Previus we are using EasyExtendsBundle to do it in `Application\Sonata\UserBundle\Entity\User`.
IMO in UserBundle v4 we should provide recipies with default entities. Using `Sonata\UserBundle\Entity\BaseUser` will be also allowed.

Reply:
So we shouldn't use `'App\Entity\SonataUserUser'` in the config
But `'Sonata\UserBundle\Entity\User'`

Reply:
I recheck it. It is not possible to use `Sonata\UserBundle\Entity\BaseUser`. I think it is not possible to provide User class in easy way. We should not provide User identifier by own and can not use two User classes for one entity. So we must keep `App\Entity\SonataUserUser`

Reply:
Why ?

You still provide the classes BaseUser and a BaseGroup. What's wrong with them.

Reply:
But these classes must be override to configure identifier for them. Also in other bundles we are using `App\Entity\Sonata*`

https://github.com/sonata-project/SonataPageBundle/blob/3.x/docs/reference/advanced_configuration.rst
https://github.com/sonata-project/SonataArticleBundle/blob/1.x/docs/reference/configuration.rst
https://github.com/sonata-project/SonataMediaBundle/blob/3.x/docs/reference/advanced_configuration.rst
https://github.com/sonata-project/SonataNewsBundle/blob/3.x/docs/reference/advanced_configuration.rst

Reply:
Ok, so we should add a note about the interface that the customUser/Group need to implements and be sure it's clear we provide a Base class to help developers.

Reply:
I did not work on upgrade note jet. I waiting for appove current change. Then I will update note based on approved changes.